### PR TITLE
Adds Baseballs

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Adds baseballs to the game (There is a baseball field in the holodeck).
You can now bat thrown objects with the baseball bat to launch them away at high speed.
https://streamable.com/471jvv (baseball is a boomerang here because otherwise this would have been impossible to test singleplayer)

## Why It's Good For The Game

it could be fun to have a game of baseball, and people trying to bat an item thrown at them sounds funny 
image

## Changelog

:cl: Fikou, sprite by Mooster
add: Baseballs are now available in the Baseball Field on the Holodeck.
add: Baseball Bats can now hit thrown objects mid-air to send them back.
/:cl:

### CODE REVIEW

**Comment by calliophis-maculiceps** [comment]
can't wait to bat chemical grenades back to sender

**Comment by Edelor68** [comment]
Do batted items do more damage than thrown items?

**Comment by Fikou** [comment]
> Do batted items do more damage than thrown items?

you can only bat away thrown items, and it doubles their throw force temporarily

**Comment by necromanceranne** [comment]
Ultimate gaming: batting bolas.

**Comment by timothymtorres** [comment]
♫♫♫
_Take me out to the ball game
Take me out with the crowd
Buy me some peanuts and crackerjacks
I don’t care if I never get back
Let me root, root, root
For the home team
If they don’t win it’s a shame
Aahh.
For it’s one,
Two,
Three strikes you’re out
At the old ball game_
♫♫♫

This is great and long overdue.  SS13 has been here for 20 years without any kind of baseball reference.  It's about damn time!  +1

**Comment by Farquaar** [comment]
https://wiki.teamfortress.com/wiki/Sandman

**Comment by Wallemations** [comment]
Oh my god the ablative bat actually does what it's comment said it would do for years

**Comment by TemporalOroboros** [comment]
Does this have a cooldown or can you ping-pong something between two people with bats if they are good at clicking?
Given that the throwforce doubles with each iteration...

**Comment by Shadowflame909** [comment]
when the immovable rod is coming for you but you have a baseball bat

File: code/game/objects/items/weaponry.dm [hunk 1]
```diff
@@ -732,28 +735,66 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+	addtimer(CALLBACK(src, .proc/launch_back, target, user, return_to_sender, datum_throw_speed), 0.5 SECONDS)
+	return TRUE
+
+/obj/item/melee/baseball_bat/proc/launch_back(atom/movable/target, mob/living/user, turf/target_turf, datum_throw_speed)
+	playsound(target, 'sound/magic/tail_swing.ogg', 50, TRUE)
+	REMOVE_TRAIT(user, TRAIT_IMMOBILIZED, type)
+	target.mouse_opacity = initial(target.mouse_opacity)
+	target.add_filter("baseball_launch", 3, motion_blur_filter(1, 3))
+	target.throwforce *= 2
+	target.throw_at(target_turf, get_dist(target, target_turf), datum_throw_speed + 1, user, callback = CALLBACK(src, .proc/on_hit, target))
```
Comment (ATH1909) [inline code/game/objects/items/weaponry.dm:780]: will on_hit() always get called? even if, say, the baseball bat gets deleted or the ball never ends up hitting anything? in other words, are we 10,000% sure that people can't use this to give an object an arbitrarily high throwforce?

**Comment by Fikou** [comment]
i will make this mechanic easier its a bit bs to hit it currently

**Review by Imaginos16 — APPROVED** [review]
Fine for the purposes of the PR

**Pull request merged**